NPM packages and test vendor fixes

jgonggrijp opened the pull request:
- Adds a package lock file (`yarn.lock`).
- Removes the `test/vendor` directory in favor or the `node_modules`.
- Updates the test code to use namespaced `QUnit.test` instead of global `test`, etcetera.
- Upgrades the outdated packages that remained after #221.

Together, these changes should fix #218.

All tests still pass, except for the `underscore.util.strings:toQuery:"query serialization matchs jQuery.param()"` test. I upgraded jQuery from version 1.7 to 3.5, and in the meanwhile it changed from urlencoding spaces with a `+` to encoding them with the `%20` escape sequence.

Should I realign `_.toQuery` with jQuery 3? Also, are you OK with using Yarn as the package manager?

joshuacc commented:
Yes, I would update to match jQuery 3. Just need to note that it might be a breaking change for some people. Low probability, I think.

I'm fine with Yarn. But need to make sure the docs for contributing mention it, and which version of Yarn to use.

jgonggrijp commented:
@joshuacc Thanks for your review! I have addressed your comments, though I have not referenced an explicit version number for Yarn (yet). [This documentation](https://yarnpkg.com/getting-started/install) seems to be telling me that Yarn will always install globally as version 1 and then automagically turn into version 2 if a project is configured to use that. I happened to be using version 1 while generating the lockfile, so my impression is that there is no risk that people will be using the wrong version.

Even if such a thing did happen, though, it shouldn't be too disastrous. Yarn 2 is backwards-compatible (at least as far as the features used in Contrib are concerned), and users who previously worked on Contrib with Yarn 1 will be able to switch with a single command.

Please let me know whether you agree.

joshuacc commented:
Sounds good to me. 👍
